File: Mar2026/18thMarch2026/twoSum.py
def two_sum(arr, target):
    hash = {}
    for i , num in enumerate(arr):
        print("i", i)
        diff = target - num
        print("hash", hash)
        if(diff in hash):
            return [hash[diff], i]
        hash[num] = i
print(two_sum([2,7,11,15], 9))

File: May2026/2ndMay2026/twoSum.js
function twoSum(arr, target) {
  let map = new Map();
  for (let i = 0; i < arr.length; i++) {
    let diff = target - arr[i];
    if (map.has(diff)) {
      return [map.get(diff), i];
    }
    map.set(arr[i], i);
  }
}

console.log(twoSum([2, 3, 4, 5], 9));

// [2,3,4,5], 9
